packages/protocol/test-sol/voting/Election.t.sol

```diff
@@ -0,0 +1,2543 @@
+    public
+  {
+    WhenBothGroupsHaveBothPendingAndActiveVotes_WhenAccountHasVotedForMoreThanOneGroupInequallyWithDecrement();
+
+    assertEq(election.getTotalVotesByAccount(voter), remaining + value2);
+  }
+
+  function test_ShouldReduceGroup1PendingVotesTo0_WhenWeSlash1MoreVoteThanGroup1PendingVoteTotal_WhenAccountHasVotedForMoreThanOneGroupInequally()
+    public
+  {
+    WhenBothGroupsHaveBothPendingAndActiveVotes_WhenAccountHasVotedForMoreThanOneGroupInequallyWithDecrement();
+    assertEq(election.getPendingVotesForGroupByAccount(group, voter), 0);
+  }
+
+  function test_ShouldReduceGroup1ActiveVotesBy1_WhenWeSlash1MoreVoteThanGroup1PendingVoteTotal_WhenAccountHasVotedForMoreThanOneGroupInequally()
+    public
+  {
+    WhenBothGroupsHaveBothPendingAndActiveVotes_WhenAccountHasVotedForMoreThanOneGroupInequallyWithDecrement();
+    assertEq(election.getActiveVotesForGroupByAccount(group, voter), remaining);
+  }
+
+  uint256 totalRemaining;
+  uint256 group1Remaining;
+  uint256 group2TotalRemaining;
+  uint256 group2PendingRemaining;
+  uint256 group2ActiveRemaining;
+
+  function WhenWeSlashAllOfGroup1VotesAndSomeOfGroup2__WhenWeSlash1MoreVoteThanGroup1PendingVoteTotal_WhenAccountHasVotedForMoreThanOneGroupInequally()
+    public
+  {
+    WhenBothGroupsHaveBothPendingAndActiveVotes_WhenAccountHasVotedForMoreThanOneGroupInequally();
+
+    slashedValue = value + 1;
+
+    totalRemaining = value + value2 - slashedValue;
+    group1Remaining = 0;
+    group2TotalRemaining = value2 - 1;
+    group2PendingRemaining = value2 / 2 - 1;
+    group2ActiveRemaining = value2 / 2;
+
+    forceDecrementVotes2Groups(group, address(0), address(0), group2);
+  }
+
+  function test_ShouldDecrementGroup1Votes_WhenWeSlashAllOfGroup1VotesAndSomeOfGroup2__WhenWeSlash1MoreVoteThanGroup1PendingVoteTotal_WhenAccountHasVotedForMoreThanOneGroupInequally()
+    public
+  {
+    WhenWeSlashAllOfGroup1VotesAndSomeOfGroup2__WhenWeSlash1MoreVoteThanGroup1PendingVoteTotal_WhenAccountHasVotedForMoreThanOneGroupInequally();
+    assertEq(election.getTotalVotesForGroupByAccount(group, voter), group1Remaining);
+    assertEq(election.getTotalVotesForGroup(group), group1Remaining);
+    assertEq(election.getPendingVotesForGroupByAccount(group, voter), group1Remaining);
+    assertEq(election.getActiveVotesForGroupByAccount(group, voter), group1Remaining);
+  }
+
+  function test_ShouldDecrementGroup2Votes_WhenWeSlashAllOfGroup1VotesAndSomeOfGroup2__WhenWeSlash1MoreVoteThanGroup1PendingVoteTotal_WhenAccountHasVotedForMoreThanOneGroupInequally()
+    public
+  {
+    WhenWeSlashAllOfGroup1VotesAndSomeOfGroup2__WhenWeSlash1MoreVoteThanGroup1PendingVoteTotal_WhenAccountHasVotedForMoreThanOneGroupInequally();
+    assertEq(election.getTotalVotesForGroupByAccount(group2, voter), group2TotalRemaining);
+    assertEq(election.getTotalVotesByAccount(voter), totalRemaining);
```

Comment:
NIT: make this its own test

Reply:
Personally I would prefer to keep it as it is since it would lots of 2 liner tests